### ISSUE TITLE
Search spam detected issue after AirDC v4.21 "update"

### ISSUE THREAD

**Issue by ryehamstrawberry** [luadch/luadch#226]
I'm posting this bug in the event that Luadch doesn't have some sort of search spam limitation and that's what's partially responsible for the following message:
`*** Search spam detected from <IP> (<username>) (severe). The hub will be disconnected in order to preserve system resources. If you continue receiving this error, it may indicate that the hub utilizes its users in a DDoS attack.`

Ever since the latest AirDC++ v4.21 update, the previous error has been constantly reported by many users.
In some cases just the message is received but in other situations, entire hub userlists are being disconnected simultaneously.
I posted this issue on the AirDC bug tracker too because I have a feeling it's a combination of both projects not playing well with each other: [Search spam detected (False Positive Behavior)](https://github.com/airdcpp/airdcpp-windows/issues/94)

Since I have little faith the AirDC dev is going to remove or change his client's behavior, I'm hoping Luadch devs will look into a possible solution for this. 

For some reason the AirDC dev keeps updating his client to address issues that don't seem to be affecting very many users and in the process causes inconvenient client behavior for the majority of users, then blames Luadch 😂.
Remember [SECURITY WARNING: received a DSCH message that should have been sent to a different user.](https://github.com/airdcpp/airdcpp-windows/issues/76)?
I feel sorry that AirDC "updates" address none of the community's main feature requests and bugs and instead addresses problems no one is experiencing, which just causes more work for Luadch.
/rant

**Comment by blasti** [comment]
Luadch has no search spam detection. It is intended for private reg only hubs where you can ban offending or abusive users.

Right now I don't have the ressources to implment spam detection.

**Comment by maksis** [comment]
> It is intended for private reg only hubs where you can ban offending or abusive users.

I don’t think this issue is related to abusive users. It occurs when the connection between the client and the hub has been inoperable for several minutes (potentially even dozens of minutes), causing all accumulated protocol messages to be received simultaneously once the connection is restored.

The hub can block a large number of messages sent by a single user, which would resolve the majority of spam warnings. However, it cannot prevent individual users with connection issues from receiving large chunks of data after such blackouts. I’m not sure if this can be resolved without implementing some kind of ping-pong mechanism between the client and the hub, allowing both parties to detect when messages are not being transmitted properly.

**Comment by 4C3T** [comment]
I'm thinking that if the hubsoft can detect a user who's doing this alleged spam that's so "harmful" it shuts down the hub to prevent it, why not simply boot the user. Say that this flood some buffers that will cause a crash or whatever, a restart or whatever it now do when this occurs, why not add a time ban for xx minutes/hours/days for the user causing it? If it happens once, the hub either boots the user, or restarts, bans the user.

Could not this be a viable solution, or am I missing something here?
